test/fixtures/es-module-loaders/js-loader.mjs

```diff
@@ -1,9 +0,0 @@
-export function getFormat(url, context, defaultGetFormat) {
-  // Load all .js files as ESM, regardless of package scope
-  if (url.endsWith('.js')) {
-    return {
-      format: 'module'
-    }
-  }
-  return defaultGetFormat(url, context, defaultGetFormat);
-}
```

Comment:
Instead of removing a test loader like this and its tests, let's rather keep the tests and rewrite the loader using the new pattern.

Reply:
[hooks-custom.mjs](https://github.com/nodejs/node/pull/37468/files/e7422eb13ca4df47df66e86803c3e417065c067c#diff-3a607d0f76bd2ac5c7703223f1e4413cde25cb367c1944da51099cc54117fbd9R10) has this and `getFormat` isn't directly accessible anymore (it's wrapped by `load`).

`test-esm-loader.mjs` now (as of this PR) contains all cases relevant to EMSLoader and custom hooks (so you don't have to go spelunking through hundreds of test files for related cases). I (just now) included a case that covers this specific scenario: [`import('../fixtures/es-module-loaders/js-as-esm.js')`](https://github.com/nodejs/node/pull/37468/files#diff-4a8a0345f25cf8762e6ffd591dea32050b6b51dc0e02638247ec10595fe19f29R36)

Does that work for you?

Reply:
Sure, as long as we have the same overall coverage cases.